tests/integration/render.rs

```diff
@@ -172,54 +176,60 @@ fn pipe_output_to_a_tool_that_exists() -> anyhow::Result<()> {
         "image: \"nginx:latest\"\n          imagePullPolicy: IfNotPresent\n"
     );
 
-    drop_temp_folder(&format!("{}", config.parent().unwrap().to_string_lossy()))?;
-
     Ok(())
 }
 
 #[test]
 fn pipe_output_to_multiple_tools() -> anyhow::Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
 
-    let config = generate_config().unwrap();
+    let config = Config::new()?;
 
     cmd.arg("render")
         .arg("--pipe=grep 'image'")
         .arg("--pipe=grep 'imagePullPolicy'")
-        .arg(&config);
+        .arg(&config.path);
 
     cmd.assert().success();
 
     let edge_rendered_output = format!(
         "{}/manifests/edge-eu-w4/my-app/manifest.yaml",
-        config.parent().unwrap().to_string_lossy()
+        config.temp_dir.to_string_lossy()
     );
 
-    let mut yaml = std::fs::File::open(edge_rendered_output).unwrap();
+    let mut yaml = std::fs::File::open(edge_rendered_output)?;
 
     let mut contents = "".to_string();
     yaml.read_to_string(&mut contents)?;
 
     // assert that the output contains only imagePullPolicy related content
     assert_eq!(contents.trim_start(), "imagePullPolicy: IfNotPresent\n");
 
-    drop_temp_folder(&format!("{}", config.parent().unwrap().to_string_lossy()))?;
-
     Ok(())
 }
 
 #[test]
 fn pipe_output_to_a_tool_that_doesnt_exist() -> anyhow::Result<()> {
     let mut cmd = Command::cargo_bin(BIN_NAME)?;
+    let config = Config::new()?;
 
-    let config = generate_config().unwrap();
-
-    cmd.arg("render").arg("--pipe=xyz 'image'").arg(&config);
+    cmd.arg("render")
+        .arg("--pipe=xyz 'image'")
+        .arg(&config.path);
 
     // the binary execution should fail because xyz tool doesn't exist.
     cmd.assert().failure();
 
-    drop_temp_folder(&format!("{}", config.parent().unwrap().to_string_lossy()))?;
+    Ok(())
+}
+
+#[test]
+fn render_multiple_files() -> anyhow::Result<()> {
```

Comment:
@Kenec and here goes the multi-file test case to assert the feature works as expected.